### PR TITLE
lib/posix-environ: Add comments for posix-environ

### PR DESCRIPTION
**Pull request by DeliaPavel**
### Prerequisite checklist

 - [ ] Read the [contribution guidelines](https://unikraft.org/docs/contributing/) regarding submitting new changes to the project;
 - [ ] Tested your changes against relevant architectures and platforms;
 - [ ] Ran the [`checkpatch.uk`](https://github.com/unikraft/unikraft/blob/staging/support/scripts/checkpatch.uk) on your commit series before opening this PR;
 - [ ] Updated relevant documentation.

### Base target

 - Architecture(s): [e.g. `x86_64` or N/A]
 - Platform(s): [e.g. `kvm`, `xen` or N/A]
 - Application(s): [e.g. `app-python3` or N/A]

### CODE REVIEW

File: lib/posix-environ/environ.h [hunk 1]
```diff
@@ -9,9 +9,34 @@
...
+ *   The name of the environment variable to be set
+ * @param l
+ *   The length of the environment variable name
+ * @param r
+ *   The value of the environment variable to be set
+ * @return
+ *   0 in case of success, -1 in case of error
+ */
 int __putenv(char *s, size_t l, char *r);
+/**
```
Comment (StefanJum) [inline lib/posix-environ/environ.h]:
```suggestion
int __putenv(char *s, size_t l, char *r);

/**
```

**Review by StefanJum — CHANGES_REQUESTED** [review]
One small comment @DeliaPavel, all good after that.

**Review by StefanJum — APPROVED** [review]
Thank you @DeliaPavel 
Approved-by: Stefan Jumarea <stefanjumarea02@gmail.com>

**Review by razvand — APPROVED** [review]
Reviewed-by: Razvan Deaconescu <razvand@unikraft.io>

**Pull request merged**